<launch>
  <arg name="x" default="0" />
  <arg name="y" default="0" />
  <arg name="z" default="1.0" />
  <arg name="yaw" default="0" />
  <arg name="joystick" default="true" />

  <!-- Configuration of Jackal which you would like to simulate.
       See jackal_description for details. -->
  <!-- <arg name="config" default="base" /> -->

  <!-- Load Jackal's description, controllers, and teleop nodes. -->
  <include file="$(find amiga_description)/launch/amiga_description.launch">
    <!-- <arg name="config" value="$(arg config)" /> -->
  </include>
  
  <!-- Load controller. -->
  <include file="$(find amiga_control)/launch/amiga_control.launch" />

  <!-- <include file="$(find jackal_control)/launch/teleop.launch">
    <arg name="joystick" value="$(arg joystick)" />
  </include> -->

  <!-- Spawn Jackal -->
  <arg name="sdf_robot_file" default="$(find amiga_gazebo)/models/farmland_1/farmland_1.sdf"/>
  <arg name="robot_name" default="FramLand"/>

  <node name="$(arg robot_name)_spawn_urdf" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-file $(arg sdf_robot_file) -sdf -x 0 -y 0 -z 0 -R 0 -P 0 -Y 0 -model $(arg robot_name)"/>

  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
	      args="-urdf -model amiga -param robot_description -x $(arg x) -y $(arg y) -z $(arg z) -R 0 -P 0 -Y $(arg yaw)" />
        
  <node pkg="interactive_marker_twist_server" type="marker_server" name="twist_marker_server">
    <param name="link_name" value="base_link" />
  </node>

</launch>
